ecl/hql/hqlmanifest.cpp

```diff
@@ -198,6 +198,64 @@ void ResourceManifest::loadResource(const char *filepath, MemoryBuffer &content)
     read(fio, 0, (size32_t) f->size(), content);
 }
 
+void addSecretToJfrog(IPropertyTree &item)
+{
+    StringBuffer password;
+    const char *user = item.queryProp("@jfrogUser");
+    getSecretValue(password, "jfrog", user, "password", true);
+    password.stripChar('\n').stripChar(' ').str(); // Strip newlines/whitespaces in case the secret has been entered by hand.
+
+    StringBuffer fileContents;
+    const char *server = item.queryProp("@jfrogServer");
+    fileContents.setf("{\n\
+    \"servers\": [\n\
+        {\n\
+        \"url\": \"%s/\",\n\
+        \"artifactoryUrl\": \"%s/artifactory/\",\n\
+        \"distributionUrl\": \"%s/distribution/\",\n\
+        \"xrayUrl\": \"%s/xray/\",\n\
+        \"missionControlUrl\": \"%s/mc/\",\n\
+        \"pipelinesUrl\": \"%s/pipelines/\",\n\
+        \"user\": \"%s\",\n\
+        \"password\": \"%s\",\n\
+        \"serverId\": \"HPCC\",\n\
+        \"isDefault\": true\n\
+        }\n\
+    ],\n\
+    \"version\": \"6\"\n}\n", server, server, server, server, server, server, user, password.str());
+
+    StringBuffer configPath(getenv("HOME"));
```

Comment:
Use getHomeDir() in jmisc.hpp - which will also work on windows.
I assume this is a location defined by jfrog.

Reply:
Fixed

Reply:
Yes, this is the location where the JFrog CLI expects the config file to be if there are no credentials passed through the command line.